nemo/collections/nlp/data/language_modeling/megatron/indexed_dataset.py

```diff
@@ -488,25 +488,33 @@ def __getitem__(self, i):
         def __len__(self):
             return self._len
 
-    def __init__(self, path, skip_warmup=False):
+    def __init__(self, path, skip_warmup=False, delay_data_mmap=False):
         super().__init__()
 
         self._path = None
         self._index = None
         self._bin_buffer = None
+        self._delay_data_mmap = delay_data_mmap
+        self._skip_warmup = skip_warmup
 
-        self._do_init(path, skip_warmup)
+        self._do_init(path, skip_warmup, delay_data_mmap)
 
     def __getstate__(self):
         return self._path
 
     def __setstate__(self, state):
         self._do_init(state)
 
-    def _do_init(self, path, skip_warmup=True):
+    def _do_init(self, path, skip_warmup=True, delay_data_mmap=False):
         self._path = path
         self._index = self.Index(index_file_path(self._path), skip_warmup)
 
+        if not delay_data_mmap:
+            self._create_data_mmap(skip_warmup)
+        else:
+            logging.info("    skip creating data numpy buffer of mmap...")
```

Comment:
remove whitesapce

Reply:
Added this whitespace to align with other print statements: https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/data/language_modeling/megatron/indexed_dataset.py#L511-L515 
I can remove it if it's not needed. Please let me know.

Reply:
Unless there's a reason for it, please remove. Old statements may have just been copied over.